test/integration/cli-triton-tags.test.js

```diff
@@ -0,0 +1,159 @@
+                        'triton.cns.disable': 'true'
+                    }
+                }
+            }
+        });
+    });
+
+    tt.test('  check triton.cns.disable tag on VM', function (t) {
+        vm.get(t, {
+            id: containerId,
+            partialExp: {
+                tags: {
+                    'sdc_docker': true,
+                    // VMAPI tags can be boolean, hence `true` here.
+                    'triton.cns.disable': true
+                }
+            }
+        });
+    });
+
+
+    /*
+     * Next, test that we can update "triton.*" tags on the VM (via VMAPI)
+     * and have those updates show in `docker inspect ...`.
+     */
+    tt.test('  add/update triton.* tags via VMAPI', function (t) {
+        vm.addTags(t, {
+            id: containerId,
+            tags: {
+                'triton.cns.disable': false,
+                'triton.cns.services': 'foo,bar'
+            }
+        });
+    });
+
+    // Updating tags is async, so need to wait for them to propogate.
+    tt.test('  wait for VMAPI tag updates', function (t) {
+        vm.waitForTagUpdate(t, {
+            id: containerId,
+            tags: {
+                'triton.cns.disable': false,
+                'triton.cns.services': 'foo,bar'
+            }
+        });
+    });
+
+    tt.test('  check VMAPI tag updates on container labels', function (t) {
+        cli.inspect(t, {
+            id: containerId,
+            partialExp: {
+                Config: {
+                    Labels: {
+                        'com.joyent.package': '*',
+                        // Note that Docker labels are *strings*, hence 'true'.
+                        'triton.cns.disable': 'false',
+                        'triton.cns.services': 'foo,bar'
+                    }
+                }
+            }
+        });
+    });
+
+
+    /*
+     * Next, sanity check that "triton.*" labels to "docker run" are
+     * validated.
+     */
+    tt.test('  invalid triton tag: triton.cns.disable=nonbool', function (t) {
+        var name = CONTAINER_PREFIX + libuuid.create().split('-')[0];
+        var args = format(
+            '--label triton.cns.disable=nonbool -d --name %s alpine sleep 3600',
+            name);
+        cli.run(t, {
+            args: args,
+            /* JSSTYLED */
+            expectedErr: 'Error response from daemon: Triton tag "triton.cns.disable" value must be "true" or "false": "nonbool"'
+        });
+    });
+
+    tt.test('  invalid triton tag: triton.bogus=foo', function (t) {
+        var name = CONTAINER_PREFIX + libuuid.create().split('-')[0];
+        var args = format(
+            '--label triton.bogus=foo -d --name %s alpine sleep 3600', name);
+        cli.run(t, {
+            args: args,
+            /* JSSTYLED */
+            expectedErr: 'Error response from daemon: Unrecognized special triton tag "triton.bogus"'
+        });
+    });
+});
+
+
+test('teardown', cli.rmAllCreated);
```

Comment:
We seem to have tested boolean and string values, should we also add a test for a triton.cns tag that has a numeric value so we test the:

```
case 'number':
```

codepath? With both a valid and invalid number and maybe a float and an empty string?